src/Extension/Traits/LocalisedCopyTrait.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+namespace TractorCow\Fluent\Extension\Traits;
+
+use SilverStripe\Versioned\Versioned;
+use TractorCow\Fluent\Extension\FluentExtension;
+use TractorCow\Fluent\Forms\CopyLocaleAction;
+
+/**
+ * Trait LocalisedCopyTrait
+ *
+ * Provides a hook which is executed when object is localised
+ * This is useful in case some action needs to be executed as a part of locale copy process
+ * examples:
+ * - related object needs to be duplicated into the new locale
+ * - event needs to be dispatched when page is localised
+ * - localised copy of a page needs to be executed outside of usual flow
+ *
+ * @see CopyLocaleAction
+ */
+trait LocalisedCopyTrait
+{
+    /**
+     * Flag indicating if the object is going through copy to locale action
+     * this is needed to determine if we need to perform localised copy
+     * this is an in-memory only field
+     *
+     * @var bool
+     */
+    private $localisedCopyActive = false;
+
+    public function getLocalisedCopyActive(): bool
+    {
+        return $this->localisedCopyActive;
+    }
+
+    public function setLocalisedCopyActive(bool $active): self
+    {
+        $this->localisedCopyActive = $active;
+
+        return $this;
+    }
+
+    /**
+     * Provides a safe way to temporarily alter the global flag
+     * useful for unit tests
+     *
+     * @param callable $callback
+     * @return mixed
+     */
+    public function withLocalisedCopyState(callable $callback)
```

Comment:
When is this used outside of tests? I am worried about test-only code that could be testing a use case that doesn't reflect day to day usage.

Reply:
This is not test-only code as there are other use cases for it. For example data migration from third party (page content is recreated during update). I'll update the docblock.